Resources/config/doctrine-mapping/User.mongodb.xml

```diff
@@ -2,46 +2,43 @@
 <doctrine-mongo-mapping xmlns="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping"
                         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                         xsi:schemaLocation="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping
-                        http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping.xsd">
+                    http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping.xsd">
 
     <mapped-superclass name="FOS\UserBundle\Model\User" collection="fos_user_user">
 
-        <field name="username" type="string" />
+        <field name="username" field-name="username" type="string" />
```

Comment:
There's no need to specify these manually, as ODM will assume the field name to be the same as the database name by default. Since #2934 contains the same set of changes, I suggest this PR be closed.